google/cloud/pubsublite/internal/resumable_async_streaming_read_write_rpc.h

```diff
@@ -495,7 +495,10 @@ class ResumableAsyncStreamingReadWriteRpcImpl
     auto stream = std::move(*start_initialize_response);
     std::unique_lock<std::mutex> lk{mu_};
     if (stream_state_ == State::kShutdown) {
-      stream->Finish().then([](future<Status>) {});
+      // We need to call `Finish()` to satisfy the gRPC requirements. We also
+      // need to extend the lifetime to `stream` until `Finish()` returns.
+      auto extended = std::shared_ptr<StreamType>(std::move(stream));
+      extended->Finish().then([extended](auto) {});
```

Comment:
It looks like this will hit `ref_count == 2`, and that we could avoid that, but surely not worth worrying about in the big picture.

Reply:
Ack